edfi-repo-auditor/edfi_repo_auditor/auditor.py

```diff
@@ -45,18 +58,39 @@ def run_audit(config: Configuration) -> None:
         score = get_result(results, auditing_rules["rules"])
         logger.debug(f"Rules to follow: {auditing_rules}")
 
-        report[repo] = {
+        report_json[repo] = {
             "score": score,
-            "result": "OK" if score > auditing_rules["threshold"] else "Action required",
-            "description": results
+            "result": "OK"
+            if score > auditing_rules["threshold"]
+            else "Action required",
+            "description": results,
         }
 
+        report_data.append(
+            {
+                "repository": repo,
+                "result": "OK"
+                if score > auditing_rules["threshold"]
+                else "Action required",
+                **actions,
+                **file_review,
+                **repo_config,
+            }
+        )
+
+    df = pd.DataFrame(report_data)
+
     if config.save_results is True:
-        save_to_file(report, config.file_name)
+        save_to_json(report_json, config.file_name)
```

Comment:
In the future, we could try to use only the panda's DataFrame and convert the results to JSON and HTML since there's some code repetition when generating, and we avoid building two objects